fix(daemon/server): break out of the stream-events handler

Profpatsch opened the pull request:
If the client disengages, we need to stop the handler immediately,
otherwise we keep it running in the for-loop for all times.

This fixes a problem where any call of the
`lorri internal stream-events --kind snapshot` client
would drop the connection, and the handler would trace it but not stop
itself.

Fixes https://github.com/nix-community/lorri/issues/35



<!--
Explain the approach you took to resolving the issue and provide necessary context.
There is no need to go into a lot of detail here: instead, try to make each commit self-explanatory and write good commit messages.
-->

- [ ] Amended the changelog in `release.nix` (see `release.nix` for instructions)

Profpatsch committed:
fix(daemon/server): break out of the stream-events handler

If the client disengages, we need to stop the handler immediately,
otherwise we keep it running in the for-loop for all times.

This fixes a problem where any call of the
`lorri internal stream-events --kind snapshot` client
would drop the connection, and the handler would trace it but not stop
itself.

Fixes https://github.com/nix-community/lorri/issues/35

Profpatsch commented:
cc @symphorien

symphorien reviewed (approved):
fixes the issue. thanks !

Profpatsch committed:
chore(/): &PathBuf -> &Path

Make clippy happy.

There’s another thing where panic! will only accept a formatting
string in rust 2021.
